tensorflow_datasets/scripts/random_image_generator.py

```diff
@@ -0,0 +1,132 @@
+
+""" Generate random images that compress better.
+    By replacing the larger images with more compressable equivalents. """
+
+import os
+import tempfile
+import zipfile
+import tarfile
+from PIL import Image
+import numpy as np
+from absl import app
+from absl import flags
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string("dir_path",
+                    "testing/test_data/fake_examples",
+                    "path to the directory which contains files")
+
+def image_process(filepath):
+    """ Generate random images and remove noise of the
+        images to compress better.
+
+          Args:
+           filepath: path of the images to get processed
+
+    """
+    image = np.array(Image.open(filepath))
+    grey = int(hash(filepath) % 255)
+    image = np.ones_like(image)*grey
+    image = Image.fromarray(image)
+    image = image.convert('RGB')
+    image.save(filepath)
+
+
+
+def rewrite_zip(_dir, path):
+    """ Rewrite the older .zip files into new better compressed one
+
+            Args:
+                _dir: directory path which contain zip compressed file
+                path: path from directory to file
+
+    """
+    # creating a temporary file to store images
+    with tempfile.TemporaryDirectory(dir=_dir) as temp:
+
+        # Extraction of compressed .zip file
+        with zipfile.ZipFile(path, 'r') as _zp:
+            _zp.extractall(path=temp)
+
+        process_dir(temp)  # Image Processing
+
+        # Compressed the .zip file again
+        with zipfile.ZipFile(path, 'w') as _zp:
+            for _file_dir, _, files in os.walk(temp):
+                for _file in files:
+                    file_path = os.path.join(_file_dir, _file)
+                    _zp.write(file_path)
+
+
+def rewrite_tar(_dir, path):
+    """ Rewrite the older .tar file into new better compressed one.
+        Compression formats supports by this method (.tar.gz, .tgz, .tar.bz2)
+
+            Args:
+              _dir: directory path which contain tar compressed file
+              path: path from directory to file
+
+    """
+    # Create a tempfile to store the images contain noise
+    with tempfile.TemporaryDirectory(dir=_dir) as temp:
+
+        #Checking the extension of file to be extract
+        ext_list = ["gz", "tgz", "bz2"]
+        read_ext = path.split('.')[-1]
```

Comment:
Rather than recomputing twice the extension. How about something like:

```
tar_ext = path.split('.')[-1]
with tarfile.open(path, _make_tar_mode('r', tar_ext)) as tar:
  ...

with tarfile.open(path, _make_tar_mode('w', tar_ext)) as tar:
  ...
```

Reply:
Is this a method _make_tar_mode()

Reply:
Yes, basically create a new private function `_make_tar_mode` to reuse the logic between `r` and `w`. As the function shouldn't be in other files, we mark it as private (starts with `_`)

Reply:
but instead of making a separate function for this we can also did this as ('r'+tar_ext) for read and ('w' + tar_ext) for write.

Reply:
Yes, this works too